src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -526,23 +530,46 @@ private RecommendedSymbols GetSymbolsOffOfExpressionInConstantPattern(Expression
             if (originalExpression is null)
                 return default;
 
-            var boundSymbol = _context.SemanticModel.GetSymbolInfo(originalExpression, _cancellationToken).Symbol;
+            var semanticModel = _context.SemanticModel;
+            var boundSymbol = semanticModel.GetSymbolInfo(originalExpression, _cancellationToken);
+
+            if (boundSymbol.Symbol is not INamespaceOrTypeSymbol namespaceOrType)
+            {
+                // Likely a Color Color case, so we reinterpret the bound symbol into a type
+                if (originalExpression is IdentifierNameSyntax identifier)
+                {
+                    var reinterpretedBinding = semanticModel.GetSpeculativeSymbolInfo(identifier.SpanStart, identifier, SpeculativeBindingOption.BindAsTypeOrNamespace);
+                    var reinterpretedSymbol = reinterpretedBinding.GetAnySymbol();
+
+                    // The reinterpretation must be a type
+                    if (reinterpretedSymbol is not INamedTypeSymbol reinterprettedNamedType)
+                        return default;
+
+                    var expression = originalExpression.WalkDownParentheses();
+
+                    return GetSymbolsOffOfBoundExpressionWorker(reinterpretedBinding, originalExpression, expression, reinterprettedNamedType, false, false);
+                }
 
-            if (boundSymbol is not INamespaceOrTypeSymbol namespaceOrType)
                 return default;
+            }
 
             var containingType = _context.SemanticModel.GetEnclosingNamedType(_context.Position, _cancellationToken);
             if (containingType == null)
                 return default;
 
-            // The RHS of an `is` pattern may only include qualifications to
+            var symbols = namespaceOrType.GetMembers().WhereAsArray(FilterMember);
+            return new RecommendedSymbols(symbols);
+
+            // A constant pattern may only include qualifications to
             // - namespaces (from other namespaces or aliases),
             // - types (from aliases, namespaces or other types),
             // - constant fields (from types)
-            // Methods, properties, events, non-constant fields etc. are excluded since they may not be present in the RHS of an `is` pattern
-            var symbols = namespaceOrType.GetMembers()
-                .WhereAsArray(s => s is INamespaceOrTypeSymbol or IFieldSymbol { IsConst: true } && s.IsAccessibleWithin(containingType));
-            return new RecommendedSymbols(symbols);
+            // Methods, properties, events, non-constant fields etc. are excluded since they are not constant expressions
+            bool FilterMember(ISymbol symbol)
+            {
+                return symbol is INamespaceOrTypeSymbol or IFieldSymbol { IsConst: true }
```

Comment:
what about a constant local?  can you add  a test for that?

Reply:
Local symbols are not contained within a type or namespace, so we will never encounter those through the filtering. However, we should have tests to make sure they are not overlooked at.

Reply:
Added tests showing the behavior in locals